mettle/src/stdapi/audio/linux_output.c

```diff
@@ -0,0 +1,66 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <mettle.h>
+#include "channel.h"
+
+#include "output.h"
+
+typedef struct context {
+    size_t size;
+    void *buffer;
+} context;
+
+int new_audio_file(struct tlv_handler_ctx *tlv_ctx, struct channel *c)
+{
+    context *ctx = malloc(sizeof(context));
+    if (ctx == NULL) {
+        return -1;
+    }
+
+    ctx->buffer = NULL;
+    ctx->size = 0;
+
+    channel_set_ctx(c, ctx);
+    return 0;
+}
+
+ssize_t write_audio_file(struct channel *c, void *buf, size_t len)
+{
+    context *ctx = channel_get_ctx(c);
+
+    ctx->size += len;
+    ctx->buffer = realloc(ctx->buffer, ctx->size);
+
+    if (ctx->buffer == NULL) {
+        return -1;
+    }
+
+    memcpy(ctx->buffer + (ctx->size - len), buf, len);
+
+    return len; // On success return the number of bytes written
+}
+
+int terminate_audio_file(struct channel *c)
+{
+    // File was uploaded, play it and free the buffer
+    context *ctx = channel_get_ctx(c);
+
+    // Open pipe, we will send the content to the aplay command
+    // Similar to `cat test.wav | aplay`
+    FILE *pipe_fp = popen("aplay", "w");
```

Comment:
we should suppress stdout, so maybe `aplay -q`?

Reply:
Should be fixed :wink: